### PR TITLE
feat(org): improve organization invitation UX

### PR DESCRIPTION
**Pull request by daryllimyt**
## Summary

This PR improves the organization invitation experience with better UX and error handling.

### Changes
- Add "Copy invitation link" button to the invitations table
- Improve invitation acceptance flow with better error messages
- Add new API endpoint to get invitation link

### Stack
1. #1972 - Auth foundation (base)
2. **This PR** - Org invitations (parallel to #1974, #1975)

## Test plan
- [ ] Test copying invitation links from the invitations table
- [ ] Test invitation acceptance flow
- [ ] Verify error messages are clear for edge cases (expired, revoked, etc.)

---

> [!NOTE]
> **Medium Risk**
> Touches invitation acceptance and membership creation logic, including new validation and atomic updates; bugs here could block onboarding or incorrectly accept/reject invites.
> 
> **Overview**
> Adds an admin-only endpoint `GET /organization/invitations/{invitation_id}/token` (and regenerated frontend client/types) so privileged users can generate shareable invitation URLs.
> 
> Updates the invitations table UI to include **Copy invitation link** for pending invites, copying an accept URL to the clipboard with success/failure toasts.
> 
> Refactors invitation acceptance to not require organization context by switching the route dependency to `AuthenticatedUserOnly` and introducing `accept_invitation_for_user()` which validates email/expiry, performs an atomic status update to avoid race conditions, and logs audit attempt/success/failure. The public “get invitation by token” handler now queries the invitation directly instead of using `OrgService`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4f5ba717f1518839cfa401c02074cc4fa74c7b35. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by daryllimyt** [comment]
* **#1973** <a href="https://app.graphite.com/github/pr/TracecatHQ/tracecat/1973?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/TracecatHQ/tracecat/1973?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1975** <a href="https://app.graphite.com/github/pr/TracecatHQ/tracecat/1975?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1974** <a href="https://app.graphite.com/github/pr/TracecatHQ/tracecat/1974?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1972** <a href="https://app.graphite.com/github/pr/TracecatHQ/tracecat/1972?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: tracecat/organization/router.py [hunk 1]
```diff
@@ -256,24 +262,105 @@ async def revoke_invitation(
...
+            role=invitation.role,
+        )
+        session.add(membership)
+
+        # Mark invitation as accepted
+        invitation.status = InvitationStatus.ACCEPTED
+        invitation.accepted_at = datetime.now(UTC)
+
+        # Single atomic commit for both membership and invitation update
+        await session.commit()
```
Comment (jordan-umusu) [inline tracecat/organization/router.py]: ideally move business logic to the service layer

**Review by jordan-umusu — CHANGES_REQUESTED** [review]
LGTM just small concern

**Review by jordan-umusu — APPROVED** [review]

**Comment by daryllimyt** [comment]
### Merge activity

* **Jan 30, 2:48 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/TracecatHQ/tracecat/1973).
* **Jan 30, 2:50 AM UTC**: [Graphite](https://app.graphite.com/github/pr/TracecatHQ/tracecat/1973) rebased this pull request as part of a merge.
* **Jan 30, 2:50 AM UTC**: @daryllimyt merged this pull request with [Graphite](https://app.graphite.com/github/pr/TracecatHQ/tracecat/1973).
